chat-widget/src/common/telemetry/loggers/appInsightsLogger.ts

```diff
@@ -0,0 +1,91 @@
+import { ApplicationInsights } from "@microsoft/applicationinsights-web";
+import { IChatSDKLogger } from "../interfaces/IChatSDKLogger";
+import { LogLevel, TelemetryEvent, TelemetryInput } from "../TelemetryConstants";
+import ScenarioMarker from "../ScenarioMarker";
+import { TelemetryHelper } from "../TelemetryHelper";
+
+export const appInsightsLogger = (appInsightsKey: string, disableCookiesUsage: boolean): IChatSDKLogger => {
+
+    let appInsights: ApplicationInsights | null = null;
+
+    const logger = (): ApplicationInsights | null => {
+        if (!appInsights && appInsightsKey) {
+            try {
+
+                const config = {
+                    ...appInsightsKey.includes("IngestionEndpoint") ? { connectionString: appInsightsKey } : { instrumentationKey: appInsightsKey },
+                    disableCookiesUsage: disableCookiesUsage,
+                };
+                
+                // Initialize Application Insights 
+                appInsights = new ApplicationInsights({ config });
+                appInsights.loadAppInsights();
+                TelemetryHelper.logActionEvent(LogLevel.INFO, {
+                    Event: TelemetryEvent.AppInsightsInitialized,
+                    Description: "Application Insights initialized successfully."
+                });
+            } catch (error) {
+                console.error("Error initializing Application Insights: ", error);
+                TelemetryHelper.logActionEvent(LogLevel.ERROR, {
+                    Event: TelemetryEvent.AppInsightsInitFailed,
+                    Description: "Error initializing Application Insights",
+                    ExceptionDetails: error
+                });
+                return null;
+            }
+        }
+        return appInsights;
+    };
+
+    const aiLogger: IChatSDKLogger = {
+        log: (logLevel: LogLevel, telemetryInput: TelemetryInput): void => {
+            try {
+                const eventName = telemetryInput?.payload?.Event;
+                const telemetryInfo = telemetryInput?.telemetryInfo?.telemetryInfo;
+                const eventProperties: ICustomProperties = {};
+                if (telemetryInfo) {
+                    const allowedKeys = ["LogLevel", "Description", "ExceptionDetails", "ChannelId", "LCWRuntimeId", "ConversationId"];
+
```

Comment:
this is a 40 LoC method, is hard to follow , 

- use of if-else-if is considerate an anti pattern
- enforce readability
- enforce single responsability, break in small chunks with a single responsability